### DISCUSSION TITLE
indent: equivalent for indent-blankline `hide_first_space_indent_level`

### DISCUSSION THREAD

**Discussion by disrupted** [folke/snacks.nvim#337] [Q&A] [answered]
I am currently evaluating the indent module as replacement for indent-blankline. Is there an equivalent for [`hooks.builtin.hide_first_space_indent_level`](https://github.com/lukas-reineke/indent-blankline.nvim/blob/259357fa4097e232730341fa60988087d189193a/doc/indent_blankline.txt#L825) in snacks.nvim?

**Comment by disrupted** [comment]
for now my workaround is to set the highlight group for the first one to have the same foreground color as my background

<img width="639" alt="screen 2024-12-18 at 20 21 24" src="https://github.com/user-attachments/assets/e77d7509-11ed-4f58-b401-898478db3ea8" />

since the last update it should now be the first 2 hl (off by 1 mistake?)

**Reply by folke** [reply, marked as answer]
Just fixed that off-by-one. Your solution is actually pretty good and more powerful than that single option.
So I'm not going to add an option for this.

**Reply by zetashift** [reply]
Is there a way to get this working that doesn't depend on the theme having a face where the foreground color matches the background color? 

Also when moving the cursor over the 'hidden' indent guide makes it look like this:
![image](https://github.com/user-attachments/assets/805cd9a7-bd62-42bf-9385-7e265a753fc2)

However I'm not sure how to change that highlight so that it matches the cursor.

**Reply by folke** [reply]
Fix your theme? custom highlight groups? Anything works, but you'll have to put in the time to investigate and create those groups.

**Reply by zetashift** [reply]
Fair enough! Will report back if I got anything nice :D.

**Comment by agingorange** [comment]
![CleanShot 2025-06-04 at 18 59 33@2x](https://github.com/user-attachments/assets/7145ecc2-2371-4348-9eb4-12e7967442db)

I was hoping the first indent guide (column) would be hidden, but as you can see, it isn't. Is this not the way to do it? I used to use 'hooks.builtin.hide_first_space_indent_level' as mentioned in the first post.

**Reply by disrupted** [reply]
Most likely your colorscheme doesn't have the `Hidden` hl group. You'd have to define it to actually blend with the background

**Reply by agingorange** [reply]
Ah, of course. Thanks.
